uast/npath.go

```diff
@@ -0,0 +1,225 @@
+package uast
+
+import (
+	"errors"
+)
+
+//CodeReference
+//https://pmd.github.io/pmd-5.7.0/pmd-java/xref/net/sourceforge/pmd/lang/java/rule/codesize/NPathComplexityRule.html
+
+//NpathComplexity return a slice of int for each function in the tree
+func NpathComplexity(n *Node) ([]int, error) {
+	var funcs []*Node
+	var npath []int
+
+	if n.containsRol(FunctionDeclarationBody) {
+		funcs = append(funcs, n)
+	} else {
+		funcs = n.deepChildrenOfRole(FunctionDeclarationBody)
+	}
+	if len(funcs) <= 0 {
+		npath = append(npath, -1)
+		return npath, errors.New("Function declaration not found")
```

Comment:
I don't see a problem with restricting ourselves to function complexity per se (we can generalize is the concept of function to code block in the future). What I meant specifically is whether we should consider not finding a function an error, instead of just returning an empty slice, which seems consistent with a function defined as _return an slice with the npath of every function found_.